scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+
+parser = argparse.ArgumentParser(description='Script helping to update nRF Connect SDK to currently recommended version.')
+parser.add_argument("-c", "--check", help="Check if your current nRF Connect SDK version is the same as recommended one.", action="store_true")
+parser.add_argument("-u", "--update", help="Update your nRF Connect SDK to currently recommended version.", action="store_true")
+args = parser.parse_args()
+
+chip_root = os.path.dirname(os.path.abspath(__file__)).replace("scripts/setup/nrfconnect", "")
+
+def get_repository_commit_sha(repository_location):
+    command = "git -C " + repository_location + " rev-parse HEAD"
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
```

Comment:
You could use a shorter `subprocess.check_call` for the same result, right? However, consider using `subprocess.run` and adding more user friendly error message on error.

Reply:
I'm not sure how that user friendly error message should look like. In that case exception is raised on some error related with calling git process, so for example it may look like the following:

fatal: cannot change to 'x/y/z': No such file or directory
Command 'git -C x/y/z rev-parse HEAD' returned non-zero exit status 128.

There might be several reasons why it could fail, like permission denied and etc, so isn't just printing original error message good enough?

Reply:
Ok, it may be. Then I only suggest using `check_call` or `check_output` functions not to repeat what they do.